src/db/lmdb_raw_lib_example_main.cc

```diff
@@ -0,0 +1,72 @@
+// Example binary to test basic functionality of LMDB.
+//
+// It is a rewriten version of
+//   https://github.com/drycpp/lmdbxx/blob/master/example.cc
+// Example cmd:
+//   mkdir /tmp/lmdb
+//   bazel run lmdb_raw_lib_example_main -- --db_dir=/tmp/lmdb
+//
+// Example output:
+//   ====================Program Starts====================
+//   Get within a txn: 'hacker@example.org'
+//   Get commited value via a write txn: 'hacker@example.org'
+//   Readonly txn w/ cursor:
+//   key: 'email', value: 'jhacker@example.org'
+//   key: 'fullname', value: 'J. Random Hacker'
+//   key: 'username', value: 'jhacker'
+
+#include <sys/stat.h>
+
+#include <cstdio>
+#include <cstdlib>
+
+#include "absl/flags/flag.h"
+#include "lmdbxx/lmdb++.h"
+
+ABSL_FLAG(std::string, db_dir, "/tmp/lmdb",
+          "Database dir for lmdb to temporarily store data."
+          "Please create your own dir and pass a absolute path.");
+
+int main() {
+  // Create and open the LMDB environment
+  std::printf("\n====================Program Starts====================\n");
+  lmdb::env env = lmdb::env::create();
+  const uint64_t mapsize = 1UL * 1024UL * 1024UL * 1024UL;  // 1 GiB
+  env.set_mapsize(mapsize);
+  const mdb_mode_t file_open_mode = (S_IRUSR | S_IWUSR);
+  env.open(absl::GetFlag(FLAGS_db_dir).c_str(), 0, file_open_mode);
+
+  // Insert some key/value pairs and get tmp value in a write transaction.
+  lmdb::txn wtxn = lmdb::txn::begin(env);
+  lmdb::dbi dbi = lmdb::dbi::open(wtxn, nullptr);
+  dbi.put(wtxn, "username", "jhacker");
+  dbi.put(wtxn, "email", "jhacker@example.org");
+  dbi.put(wtxn, "fullname", "J. Random Hacker");
+  std::string email_val;
+  dbi.get(wtxn, "email", email_val);
```

Comment:
I'm getting a std::bad_alloc termination on this line. I think it's because the value is a char*, not a string. I sent a pull request (#11) that fixes this.

Reply:
Thanks. I'll merge this first and approve yours as well.